### PR TITLE
feat!: added support tracking external services in newrelic with tracing

### PR DESCRIPTION
**Pull request by saswatds**

### CODE REVIEW

File: packages/axios/src/index.ts [hunk 1]
```diff
@@ -1,52 +1,172 @@
...
 }
 
-export default class CortecAxios implements IModule, IRequester {
+export interface IAxios {
+  service(name: string, flags?: RequestFlags): TracedAxios;
+}
+
+export enum RequestFlags {
+  None = 0,
+  Notice4XX = 1 << 0,
```
Comment (mrbrowning-postman) [inline packages/axios/src/index.ts:33]: Why the nop shift? Just for consistency if other flags are added later?

File: packages/axios/src/index.ts [hunk 2]
```diff
@@ -1,52 +1,172 @@
...
+    this.name = 'ExternalServiceError';
+  }
+}
+
+export default class Axios implements IModule, IAxios {
   name = 'axios';
-  private instances: Map<string, Axios> = new Map();
+  private instances: Map<string, A.Axios> = new Map();
+  private services: string[];
+  private nr: INewrelic | undefined;
```
Comment (mrbrowning-postman) [inline packages/axios/src/index.ts:57]: Not sure if I'm just not up to speed on idiomatic TS this way, so maybe I'm off base, but explicitly allowing `undefined` as an inhabitant of a type that we control makes me uncomfortable. It has bad and weird semantics, could we check for it at the edges and then have this field be `nr?: INewrelic` instead, unless this is for serialization purposes?

File: packages/axios/src/index.ts [hunk 3]
```diff
@@ -1,52 +1,172 @@
...
+            'patch',
+            'head',
+            'options',
+            'postForm',
+            'putForm',
+            'patchForm',
+          ].includes(prop as string) ||
+          !nr
+        )
+          return Reflect.get(target, prop, receiver);
```
Comment (mrbrowning-postman) [inline packages/axios/src/index.ts:135]: Nit: I try to always leave style concerns to automated tools, but unbraced conditionals make me antsy. Does `prettier` enforce them when the body only has one statement?

**Review by mrbrowning-postman — APPROVED** [review]
That is magical, but looks good, with a few small notes/questions. What's the process for verifying this works as expected?

File: packages/axios/src/index.ts (hunk 1 above)
Comment (saswatds) [reply packages/axios/src/index.ts:33]: Yup.

File: packages/axios/src/index.ts (hunk 2 above)
Comment (saswatds) [reply packages/axios/src/index.ts:57]: I don't think class properties can be defined with `nr?` therefore have to use `INewrelice|undefined`.

File: packages/axios/src/index.ts (hunk 3 above)
Comment (saswatds) [reply packages/axios/src/index.ts:135]: Probably. Not sure.

**Pull request merged**